docs/utime.rst

```diff
@@ -0,0 +1,159 @@
+utime
+*****
+
+.. py:module:: utime
+
+The ``utime`` module provides functions for getting the current time and date, 
+measuring time intervals, and for delays.
+
+.. note::
+    The ``utime`` module is a MicroPython implementation of the standard Python 
+    ``time`` module. It can be imported using both ``import utime`` and 
+    ``import time``, but the module is the same.
+
+
+Functions
+=========
+
+.. method:: utime.sleep(seconds)
+
+    Sleep for the given number of seconds. Some boards may accept seconds as a 
+    floating-point number to sleep for a fractional number of seconds. Note that 
+    other boards may not accept a floating-point argument, for compatibility 
+    with them use :func:`utime.sleep_ms()` and :func:`utime.sleep_us()`
+    functions.
+
+
+.. method:: utime.sleep_ms(ms)
+
+    Delay for given number of milliseconds, should be positive or 0.
+
+
+.. method:: utime.sleep_us(us)
+
+    Delay for given number of microseconds, should be positive or 0.
+
+
+.. method:: utime.ticks_ms()
+
+    Returns an increasing millisecond counter with an arbitrary reference point, 
+    that wraps around after some value.
+
+    The wrap-around value is not explicitly exposed, but we will refer to it as 
+    *TICKS_MAX* to simplify discussion. Period of the values is 
+    *TICKS_PERIOD = TICKS_MAX + 1*. *TICKS_PERIOD* is guaranteed to be a power 
+    of two, but otherwise may differ from port to port. The same period value 
```

Comment:
We need to check whats the maximum tick value and update this paragraph with that info.

Perhaps @dpgeorge can clarify here.